examples/batched-solver/batched-solver.cpp

```diff
@@ -0,0 +1,192 @@
+    using bmtx = gko::matrix::BatchCsr<ValueType, IndexType>;
+    // The gko::solver::Cg is used here, but any other solver class can also be
+    // used.
+    using cg = gko::solver::Cg<ValueType>;
+    using rich = gko::solver::BatchRichardson<ValueType>;
+
+    // Print the ginkgo version information.
+    std::cout << gko::version_info::get() << std::endl;
+
+    if (argc == 2 && (std::string(argv[1]) == "--help")) {
+        std::cerr << "Usage: " << argv[0] << " [executor] " << std::endl;
+        std::exit(-1);
+    }
+
+    // @sect3{Where do you want to run your solver ?}
+    // The gko::Executor class is one of the cornerstones of Ginkgo. Currently,
+    // we have support for
+    // an gko::OmpExecutor, which uses OpenMP multi-threading in most of its
+    // kernels, a gko::ReferenceExecutor, a single threaded specialization of
+    // the OpenMP executor and a gko::CudaExecutor which runs the code on a
+    // NVIDIA GPU if available.
+    // @note With the help of C++, you see that you only ever need to change the
+    // executor and all the other functions/ routines within Ginkgo should
+    // automatically work and run on the executor with any other changes.
+    const auto executor_string = argc >= 2 ? argv[1] : "reference";
+    const size_type nbatch = argc >= 3 ? std::atoi(argv[2]) : 2;
+    std::map<std::string, std::function<std::shared_ptr<gko::Executor>()>>
+        exec_map{
+            {"omp", [] { return gko::OmpExecutor::create(); }},
+            {"cuda",
+             [] {
+                 return gko::CudaExecutor::create(0, gko::OmpExecutor::create(),
+                                                  true);
+             }},
+            {"hip",
+             [] {
+                 return gko::HipExecutor::create(0, gko::OmpExecutor::create(),
+                                                 true);
+             }},
+            {"dpcpp",
+             [] {
+                 return gko::DpcppExecutor::create(0,
+                                                   gko::OmpExecutor::create());
+             }},
+            {"reference", [] { return gko::ReferenceExecutor::create(); }}};
+
+    // executor where Ginkgo will perform the computation
+    const auto exec = exec_map.at(executor_string)();  // throws if not valid
+
+    // @sect3{Reading your data and transfer to the proper device.}
+    // Read the matrix, right hand side and the initial solution using the @ref
+    // read function.
+    // @note Ginkgo uses C++ smart pointers to automatically manage memory. To
+    // this end, we use our own object ownership transfer functions that under
+    // the hood call the required smart pointer functions to manage object
+    // ownership. The gko::share , gko::give and gko::lend are the functions
+    // that you would need to use.
+    auto A = share(gko::read<mtx>(std::ifstream("data/A.mtx"), exec));
+    auto b = gko::read<vec>(std::ifstream("data/b.mtx"), exec);
```

Comment:
I think for the RHS and initial guesses, since they're trivial, we should remove the mtx files and just generate them in the example. Reading the matrix is sufficient demonstration of how to read from file, I think.